go/flagutil/enum.go

```diff
@@ -0,0 +1,99 @@
+/*
+Copyright 2022 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreedto in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package flagutil
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// TODO: docs for all these.
+type StringEnum struct {
```

Comment:
TODO: decide if this is a useful construct (to cause errors to happen "for free" during `pflag.Parse` vs having to post-process parsed values in each place we care) and delete if not

Reply:
I think it's useful.